tests/utils/dockercli/volumelifecycle.go

```diff
@@ -42,9 +42,14 @@ func CreateVolumeWithOptions(ip, name, options string) (string, error) {
 // AttachVolume - attach volume to container on given host
 func AttachVolume(ip, volName, containerName string) (string, error) {
 	log.Printf("Attaching volume [%s] on VM [%s]\n", volName, ip)
-	return ssh.InvokeCommand(ip, dockercli.RunContainer+"-d -v "+volName+
-		":/vol1 --name "+containerName+
-		dockercli.TestContainer)
+	return ssh.InvokeCommand(ip, dockercli.RunContainer+" -d --rm -v "+volName+
```

Comment:
OTOH I had the same issue in reverse, my vmgroups couldn't get removed because the container was still running and the volumes attached to the VM, so a vmgroup remove with --remove-volumes failed.

I'll remove the "rm" from here and add a function that will allow launcing and removing a container and will also make a stab at putting a doc together so we have an idea how the tests must be written.